src/Cli/dotnet/commands/RestoringCommand.cs

```diff
@@ -19,6 +21,7 @@ public class RestoringCommand : MSBuildForwardingApp
             string msbuildPath = null)
             : base(GetCommandArguments(msbuildArgs, noRestore), msbuildPath)
         {
+            Task.Run(() => WorkloadManifestUpdater.BackgroundUpdateAdvertisingManifestsStaticAsync());
```

Comment:
how long does this operation actually take compare with a normal restore?

Reply:
When testing with a console app, the first build took ~5 seconds and in that time the operation was able to fully complete consistently.